Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+# ETHTOOL-OFFLOADING-SETTING.sh
+# Description:
+#    1. Run TCP traffic with iperf3 between server and client VMs.
+#    2. Take a note on the offloading state and set the offloading on (assume it was off initially) on client VM.
+#    3. Disable the VF on client VM.
+#    4. Leave the traffic running between the VMs for a while.
+#    5. Reenable the VF on client VM.
+#    6. Verify the offload state on client VM.
+########################################################################
+
+remote_user="root"
+if [ ! -e sriov_constants.sh ]; then
+    cp /${remote_user}/sriov_constants.sh .
+fi
+
+# Source SR-IOV_Utils.sh. This is the script that contains all the
+# SR-IOV basic functions (checking drivers, checking VFs, assigning IPs)
+. SR-IOV-Utils.sh || {
+    echo "ERROR: unable to source SR-IOV_Utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+function version_lt() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" != "$1"; }
+
+MIN_VERSION_SUPPORTED="6.7"
```

Comment:
This is the min distro version of test coverage. Should it be ok, or move to newer version? @haiyangz

Reply:
For syncing offloadings, you need the upstream patches

Reply:
Then this script verified the upstream patch in the system first. Otherwise, it should be aborted with Logging.

Reply:
I don't see any proposal or solution against this question, but closed it. Reopen my question. This test should fail without the upstream patch. Need to find the detection of this patch in the kernel, and then continue running.

Reply:
It's difficult to check if some patch is incorporated into a binary without source code. Check the kernel version for workaround.

Reply:
Kernel version is not guaranteed. If you find the patch, and can verify the file contents of the source code or log contents. Communicate to @haiyangz about the commit id. And you will find the pattern or keyword inside.

Reply:
Yes, the kernel version is not guaranteed. Your method is needed the **source code**, but the kernel is updated with "**-CustomKerne**l" parameter in lisav2 test. There is NO source code after update.

Reply:
Understood. This is something I can communicate with @haiyangz. I will get back to here with the solution how we validate the pre-condition.